File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/utils/Stopwatch.java
package org.jetbrains.bsp.bazel.bazelrunner.utils;

import java.time.Duration;

public class Stopwatch {
  private final long start;

  private Stopwatch() {
    start = now();
  }

  public static Stopwatch start() {
    return new Stopwatch();
  }

  public Duration stop() {
    return Duration.ofMillis(now() - start);
  }

  private long now() {
    return System.currentTimeMillis();
  }
}

File: utils/src/main/kotlin/org/jetbrains/bsp/bazel/utils/time/Stopwatch.kt
package org.jetbrains.bsp.bazel.utils.time

import java.time.Duration

class Stopwatch {

    private val start: Long

    init {
        start = now()
    }

    fun stop(): Duration = Duration.ofMillis(now() - start)

    private fun now(): Long = System.currentTimeMillis()

    companion object {
        fun start(): Stopwatch = Stopwatch()
    }
}
